CONTRIBUTING.md

```diff
@@ -0,0 +1,54 @@
+
+## The `config` Directory
+
+The `config` directory at the root of this repository contains a single `versions.json` file, which allows customisation of both the version of [`access-nri/spack`](https://github.com/ACCESS-NRI/spack) that the model will be deployed on, as well as the version of [`access-nri/spack-packages`](https://github.com/ACCESS-NRI/spack-packages) that will source the ACCESS-NRI spack packages recipes.
+
+## The PR Process
+
+1. The first step is to create a branch off the model deployment repositories `main`.
+2. [Modifications](#modifications-to-the-spackyaml-file) can be made to the `spack.yaml`/`config/versions.json` files as applicable - update versions of packages, add variants, etc.
+3. Once a commit has been made that you want to build, open a PR into `main` in the model deployment repository, and the CI/CD infrastructure will attempt to build.
+    * If there are CI errors in the build, they will be reported via GitHubs Pull Request Checks. More information on them is available by clicking the failing test.
+    * If the CI checks succeed, a bot will comment on what is being deployed on the open pull request.
+    * Finally, GitHubs Environment Deployment dialogue will show on the Pull Request as a yellow 'In Progress' label. Once it shows a green 'Active' (or grey 'Inactive') deployment, you can use the modules specified in the above comment - they should look something like `MODEL/prX-Y` for the `MODEL`s `X`th PR and `Y`th deployment.
+4. Further commits can be put onto the Pull Request to create more builds, without removing the earlier ones, allowing side-by-side testing. For example, adding an extra commit will allow use of the module `MODEL/prX-(Y+1)` once it's deployed.
+5. When the Pull Request is closed, whether it is merged or not, the Prereleases **will be removed**, but can be recreated at your leisure by installing the `spack.yaml`/`spack.lock` artifact at the specified commit on your own instance, or by creating a new Pull Request.
+6. If the Pull Request is merged, it will create an official ACCESS-NRI released build of the given model, with an associated GitHub Release.
+
+## Modifications To The `spack.yaml` File
+
+Below are some real-world examples of how to modify the `spack.yaml` file.
+
+### Example Modification Workflow of `ACCESS-OM2`s `MOM5` Package
+
+In this example, we are using the `ACCESS-OM2` deployment repositories `spack.yaml` from [this commit](https://github.com/ACCESS-NRI/ACCESS-OM2/blob/47bc7bf979c1dfa12a24272cb739117abc50d7ca/spack.yaml). We are also modifying one of ACCESS-OM2's model components, namely [ACCESS-NRI/MOM5](https://github.com/ACCESS-NRI/MOM5). This model component is available in our [`ACCESS-NRI/spack-packages`](https://github.com/ACCESS-NRI/spack-packages) as the package `mom5`.
+
+1. We clone and make our modifications to the model component `ACCESS-NRI/MOM5`, which we tag with [`dev_2024.08.14`](https://github.com/ACCESS-NRI/MOM5/releases/tag/dev_2024.08.14).
```

Comment:
Can probably get rid of the "We .." at the beginning of all the instructions.

Reply:
Also I'm not sure it would be normal to tag a development branch. I think it is more normal to use a branch name.

Reply:
Done the 'we' stuff in https://github.com/ACCESS-NRI/model-deployment-template/pull/9/commits/8bc3cf1fa5a2c6018b1298efa403d236ec9fd545

Reply:
The issue with the branch name thing is that if they want to get the latest `HEAD` changes of the component on the module, they have a either force a build with `git commit --allow-empty -m "Retriggering build of COMPONENTs BRANCH"` or a comment in the `spack.yaml` noting the commit of the component branch reference. 

Honestly, I might just add the `!redeploy` ChatOps command :D

Reply:
See https://github.com/ACCESS-NRI/build-cd/issues/173 for a potential ChatOps command that could do this.

Reply:
I'm a fan of the ChatOps option, but in the meantime we definitely don't want to force people to also add tags.